src/app/_components/ProfilePageHeader/ProfilePageHeader.tsx

```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import {
+  Avatar,
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Typography,
+  useTheme,
+} from "@mui/material";
+
+interface ProfilePageHeaderProps {
+  name: string;
+  username: string;
+  bio: string;
+  avatarUrl: string; // for demonstration purpose only
+  onEditProfile: () => void;
+}
+
+const ProfilePageHeader: React.FC<ProfilePageHeaderProps> = ({
+  name,
+  username,
+  bio,
+  avatarUrl,
+  onEditProfile,
+}) => {
+  const theme = useTheme();
+
+  const contentText = (text: string | undefined | null, maxLength: number) => {
+    if (!text) return "";
+    return text.length > maxLength
+      ? `${text.substring(0, maxLength)}...`
+      : text;
+  };
+
+  const splitTextByLength = (text: string, maxLength: number) => {
+    const textLines = [];
+    for (let i = 0; i < text.length; i += maxLength) {
+      textLines.push(text.substring(i, i + maxLength));
+    }
+    return textLines;
+  };
+
+  const styleTaggedText = (text: string) => {
+    const wordsWithTags = text.split(/(\s+)/);
+
+    return (
+      <>
+        {wordsWithTags.map((word, index) =>
+          word.startsWith("@") || word.startsWith("#") ? (
+            <span key={index} style={{ color: theme.palette.primary.main }}>
+              {word}
+            </span>
+          ) : (
+            word
+          )
+        )}
+      </>
+    );
+  };
+
+  const renderBio = (bio: string) => {
+    const maxLengthPerRow = 30;
+    const maxLimitBio = contentText(bio, 80);
+    const bioContent = splitTextByLength(maxLimitBio, maxLengthPerRow);
+
+    return bioContent.map((lineOfText, index) => (
+      <Typography key={index} sx={{ fontSize: "12px" }}>
+        {styleTaggedText(lineOfText)}
+      </Typography>
+    ));
+  };
+
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "center",
+        height: "100vh",
+      }}
+    >
+      <Card
+        sx={{
+          display: "flex",
+          flexDirection: "column",
+          alignItems: "center",
+          justifyContent: "center",
+          backgroundColor: "#FAFAFA",
+          width: "375px",
+          height: "250px",
+          p: "20px",
```

Comment:
you might want this padding on the top and bottom, but not on left and right, you should let the component stretch out